docs/source/glance.rst

```diff
@@ -0,0 +1,409 @@
+   
+   # Dump a computational graph from 'loss' variable at the first iteration
+   # The "main" refers to the target link of the "main" optimizer.
+   trainer.extend(extensions.dump_graph('main/loss'))
+   
+   trainer.extend(extensions.snapshot(), trigger=(20, 'epoch'))
+   
+   # Write a log of evaluation statistics for each epoch
+   trainer.extend(extensions.LogReport())
+   
+   # Save two plot images to the result dir
+   if extensions.PlotReport.available():
+       trainer.extend(
+           extensions.PlotReport(['main/loss', 'validation/main/loss'],
+                                 'epoch', file_name='loss.png'))
+       trainer.extend(
+           extensions.PlotReport(
+               ['main/accuracy', 'validation/main/accuracy'],
+               'epoch', file_name='accuracy.png'))
+   
+   # Print selected entries of the log to stdout
+   trainer.extend(extensions.PrintReport(
+       ['epoch', 'main/loss', 'validation/main/loss',
+        'main/accuracy', 'validation/main/accuracy', 'elapsed_time']))
+   
+   #  Run the training
+   trainer.run()
+   
+   x, t = test[np.random.randint(len(test))]
+
+   predict = model.predictor(x[None]).data
+   predict = predict[0][0]
+   
+   if predict >= 0:
+      print('Predicted Poisonous, Actual ' + ['Edible', 'Poisonous'][t[0]])
+   else:
+      print('Predicted Edible, Actual ' + ['Edible', 'Poisonous'][t[0]])
+   
+If you've worked with other neural net frameworks, some of that code may look familiar. Let's break down what it's doing.
+
+Code Breakdown
+--------------
+
+Initialization
+~~~~~~~~~~~~~~
+
+Let's start our python program. Matplotlib is used for the graphs to show training progress.
+
+.. code-block:: python
+
+   #!/usr/bin/env python
+   
+   import matplotlib
+   matplotlib.use('Agg')
+
+Typical imports for a Chainer program. :class:`~chainer.links` contain trainable parameters and :class:`~chainer.functions` do not.
+
+.. code-block:: python
+
+   import chainer
+   import chainer.functions as F
+   import chainer.links as L
+   from chainer import training
+   from chainer import datasets
+   from chainer.training import extensions
+   
+   import numpy as np
+   
+Trainer Structure
+~~~~~~~~~~~~~~~~~
+
+A :class:`chainer.trainer` is used to set up our neural network and data for training. The components of the :class:`chainer.trainer` are generally hierarchical, and are organized as follows:
+
+.. image:: ../image/glance/trainer.png
+
+Each of the components is fed information from the components within it. Setting up the trainer starts at the inner components, and moves outward, with the exception of :class:`~chainer.training.extensions`, which are added after the :class:`chainer.trainer` is defined.
+
+Dataset
+~~~~~~~
+.. image:: ../image/glance/trainer-dataset.png
+
+Our first step is to format the :mod:`~chainer.datasets`. From the raw mushroom.csv, we format the data into a Chainer :class:`~chainer.datasets.TupleDataset`.
```

Comment:
* File name seems wrong.
* Let's add a link to our `mushrooms.csv` so that users can download dataset without logging into Kaggle.